### PR TITLE
External tools import and prediction functionality

### PR DESCRIPTION
**Pull request by christopher-mohr**
- Adds functionality for importing external tools
- Fixes check for available methods and prediction process 
- Adapts channel structure, input handling 
- Adapts for different input types per sample sheet

## PR checklist

- [x] This comment contains a description of changes (with reason)
- [ ] `CHANGELOG.md` is updated
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] Documentation in `docs` is updated
- [ ] If necessary, also make a PR on the [nf-core/epitopeprediction branch on the nf-core/test-datasets repo](https://github.com/nf-core/test-datasets/pull/new/nf-core/epitopeprediction)

### CODE REVIEW

File: modules/local/get_prediction_versions.nf [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+params.options = [:]
+options        = initOptions(params.options)
+
+process GET_PREDICTION_VERSIONS {
+
+    publishDir "${params.outdir}",
+        mode: params.publish_dir_mode,
+        saveAs: { filename -> saveFiles(filename:filename, options:params.options, publish_dir:'reports', meta:[:], publish_by_meta:[]) }
+
+    conda (params.enable_conda ? "bioconda::fred2=2.0.6 bioconda::mhcflurry=1.4.3 bioconda::mhcnuggets=2.3.2" : null)
```
Comment (marissaDubbelaar) [inline modules/local/get_prediction_versions.nf:13]: Dont we want to make the versioning uniform?
I noticed that you have a mulled container for fred 2.0.7 before

File: workflows/epitopeprediction.nf [hunk 2]
```diff
@@ -37,43 +37,44 @@ def modules = params.modules.clone()
...
-merge_json_single.args          = " --single_input "
+merge_json_single.args          = ' --single_input '
 merge_json_multi.args           = " --input \$PWD "
 
-
 //
 // MODULE: Local to the pipeline
 //
-include { GET_SOFTWARE_VERSIONS } from '../modules/local/get_software_versions'                                         addParams( options: [publish_files : ['tsv':'']] )
+include { GET_SOFTWARE_VERSIONS }                                   from '../modules/local/get_software_versions'       addParams( options: [publish_files : ['tsv':'']] )
```
Comment (marissaDubbelaar) [inline workflows/epitopeprediction.nf:73]: Do we still need this module?

File: modules/local/get_prediction_versions.nf (hunk 1 above)
Comment (christopher-mohr) [reply modules/local/get_prediction_versions.nf:13]: Good point, I missed that 👍

File: workflows/epitopeprediction.nf (hunk 2 above)
Comment (christopher-mohr) [reply workflows/epitopeprediction.nf:73]: Forgot to remove this.

**Comment by marissaDubbelaar** [comment]
So the last thing is to resolve the linting issue and then it can be merged right?

**Comment by christopher-mohr** [comment]
> So the last thing is to resolve the linting issue and then it can be merged right?

We will address the linting issue in a separate PR on `dsl2`.

**Review by marissaDubbelaar — APPROVED** [review]

**Pull request merged**